File: 1-100/9.palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x == 0:
            return True

        if x < 0 or x % 10 == 0:
            return False
        
        nReverse = 0
        while x > nReverse:
            nReverse = nReverse * 10 + x % 10
            x //= 10
        
        return x == nReverse or x == nReverse // 10

File: 1-100/9.palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x == 0)
        {
            return true;
        }
        
        if (x < 0 || x % 10 == 0)
        {
            return false;
        }

        int nReverse = 0;
        while (x > nReverse)
        {
            nReverse = nReverse * 10 + x % 10;
            x /= 10;
        }

        return x == nReverse || x == nReverse / 10;
    }
};